AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -309,13 +303,25 @@ - (BOOL)startService:(Class)clazz
       // Service already works, we shouldn't send log with this service name
       return NO;
     }
-    if (service.isAppSecretRequired && ![appSecret length]) {
+    if (service.isAppSecretRequired) {
+
+      /*
+       * All services that require app secret cannot be started from library. This condition should be checked before
+       * app secret validation.
+       */
+      if (!fromApplication) {
+        MSLogError([MSAppCenter logTag], @"Cannot start service %@. The service cannot be started from a library.",
+                   clazz);
+        return NO;
+      }
 
       // Service requires an app secret but none is provided.
-      MSLogError([MSAppCenter logTag],
-                 @"Cannot start service %@. App Center was started without app secret, but the service requires it.",
-                 clazz);
-      return NO;
+      if (![self.appSecret length]) {
+        MSLogError([MSAppCenter logTag],
```

Comment:
Do we really want to log this as error? I'm wondering if we should log this as a warning.

Reply:
Also, this will change with #1033 and no longer be an error

Reply:
https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java#L731